### PR TITLE
Fix ifu cube blotting for outlier detection - JP-1161

### LINKED ISSUE #4315: IFUCube blotting for outlier detection for MIRI data is not working correctly
Issue [JP-1161](https://jira.stsci.edu/browse/JP-1161) was created by Jane Morrison:

David Law reported that after #4301 was merged there still was an error in out cube_build is blotting sky back to the detector frame for outlier detection. For a MIRI exposure there are 2 channels represented. The blotting routine should map each channel separately from the sky to the detector frame. However, in the case of MIRIFUSHORT only channel 1 was mapped back and blotted. The blotted channel 2 data is  the same as the blotted channel 1 data.

### PR DESCRIPTION
**Pull request by jemorrison**
This pr addresses #JP-1161 and closes #4315

### CODE REVIEW

**Comment by jemorrison** [comment]
@stscieisenhamer I wanted you to be aware of this change to blotting since you are working on calspec3 regression testing

File: jwst/cube_build/blot_cube_build.py [hunk 1]
```diff
@@ -46,10 +44,10 @@ def __init__(self, median_model, input_models):
         self.detector = median_model.meta.instrument.detector
 
         # information on how the IFUCube was constructed
-        self.weight_power = median_model.meta.ifu.weight_power
-        self.weighting = median_model.meta.ifu.weighting
+        # self.weight_power = median_model.meta.ifu.weight_power
```
Comment (stscieisenhamer) [inline jwst/cube_build/blot_cube_build.py]: If these lines are no longer of relevance, they should be removed completely. If there is a reason to keep them around but commented, having comments regarding to why they are commented but why they may still be of relevance.

File: jwst/cube_build/blot_cube_build.py [hunk 2]
```diff
@@ -208,24 +194,34 @@ def blot_images(self):
...
                     x_slice = np.ndarray.flatten(x_slice)
                     y_slice = np.ndarray.flatten(y_slice)
                     flux_slice = np.ndarray.flatten(self.cube_flux)
-                    # print('number of original cube values mapped',flux_slice.size)
-                    valid = ~np.isnan(y_slice)
-                    x_slice = x_slice[valid]
-                    y_slice = y_slice[valid]
-                    flux_slice = flux_slice[valid]
-                    # print('number of valid cube values mapped',flux_slice.size)
+                    # print('number of original values', flux_slice.size)
```
Comment (stscieisenhamer) [inline jwst/cube_build/blot_cube_build.py]: Regarding the commented `print` statements: Much as the previous comment, these should be either removed or clarified. Helps with readability.

**Comment by stscieisenhamer** [comment]
@jemorrison There is a failing CI check: a PEP8 issue. Once resolved, go ahead and merge.

**Pull request merged**